src/Avalonia.Native/Avalonia.Native.csproj

```diff
@@ -23,9 +25,24 @@
   <ItemGroup>
     <!--<SharpGenMapping Include="Mappings.xml" />-->
     <PackageReference Include="SharpGenTools.Sdk" Version="1.2.1" PrivateAssets="all" />
-    <PackageReference Include="SharpGen.Runtime" Version="1.2.1" />
-    <PackageReference Include="SharpGen.Runtime.COM" Version="1.2.0" />
+    <PackageReference Include="SharpGen.Runtime" Version="1.2.1" PrivateAssets="all" />
+    <PackageReference Include="SharpGen.Runtime.COM" Version="1.2.0" PrivateAssets="all" />
+    <PackageReference Include="ILRepack.Lib.MSBuild.Task" Version="2.0.18.1" PrivateAssets="all" />
     <ProjectReference Include="..\..\packages\Avalonia\Avalonia.csproj" />
     <ProjectReference Include="..\Avalonia.Dialogs\Avalonia.Dialogs.csproj" />
   </ItemGroup>
-</Project>
\ No newline at end of file
+
+  <Target Name="ILRepack" AfterTargets="Build">
+    <PropertyGroup>
+      <WorkingDirectory>$(MSBuildThisFileDirectory)bin\$(Configuration)\$(TargetFramework)</WorkingDirectory>
+    </PropertyGroup>
+    <ItemGroup>
+      <InputAssemblies Include="SharpGen.Runtime.dll;SharpGen.Runtime.COM.dll;" />
+    </ItemGroup>
+    <Message Text="MERGING: @(InputAssemblies->'%(Filename)') into %(Target.FullPath)" Importance="High" />
+    <ILRepack OutputType="$(OutputType)" MainAssembly="$(AssemblyName).dll"
+              OutputAssembly="$(AssemblyName).dll" InputAssemblies="@(InputAssemblies)"
+              WorkingDirectory="$(WorkingDirectory)" 
+              KeyFile="$(AssemblyOriginatorKeyFile)" DelaySign="True" />
```

Comment:
1) Why DelaySign?
2) Are sharpgen types hidden afterwards?

Reply:
merging sharpgen into native it self is working fine. With signing there is some issue for the build, i didn't had time to look in details and didn't spend time to investigate more